code/core/src/phone/java/com/adobe/marketing/mobile/services/NetworkService.java

```diff
@@ -54,6 +56,20 @@ class NetworkService implements Networking {
 
     @Override
     public void connectAsync(final NetworkRequest request, final NetworkCallback callback) {
+        ConnectivityManager connectivityManager =
+                ServiceProvider.getInstance().getAppContextService().getConnectivityManager();
+        if (connectivityManager != null) {
+            if (!NetworkUtils.isInternetAvailable(connectivityManager)) {
+                Log.trace(ServiceConstants.LOG_TAG, TAG, "The Android network is down.");
+                callback.call(null);
+                return;
+            }
+        } else {
+            Log.warning(
+                    ServiceConstants.LOG_TAG,
+                    TAG,
+                    "ConnectivityManager is null, it's not able to check the network condition.");
```

Comment:
Can you make it a debug message?
nit: "ConnectivityManager instance is null. Unable to the check the network condition."